### PR TITLE
fix(sdk-coin-polygon): fix send tx for polygon testnet

### PR DESCRIPTION
**Pull request by gianchandania**
WIN-2819

This PR adds a method for sdk-coin-polygon to decide if non-packed encoding needs to be used or not while building the tx TICKET: WIN-2819

### CODE REVIEW

File: modules/abstract-eth/src/lib/transactionBuilder.ts [hunk 1]
```diff
@@ -627,12 +627,19 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     }
     const chainId = this._common.chainIdBN().toString();
     // This change is made to support new contracts with different encoding type
-    const coinUsesNonPackedEncodingForTxData = this._coinConfig.features.includes(
-      CoinFeature.USES_NON_PACKED_ENCODING_FOR_TXDATA
-    );
+    const coinUsesNonPackedEncodingForTxData = this.decideIfCoinUsesNonPackedEncodingForTxData();
     return this._transfer.signAndBuild(chainId, coinUsesNonPackedEncodingForTxData);
```
Comment (mullapudipruthvik) [inline modules/abstract-eth/src/lib/transactionBuilder.ts:631]:
```suggestion
    return this._transfer.signAndBuild(chainId, this.coinUsesNonPackedEncodingForTxData());
```

File: modules/abstract-eth/src/lib/transactionBuilder.ts [hunk 2]
```diff
@@ -627,12 +627,19 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
   }
 
+  /**
+   * Decide if the coin uses non-packed encoding for tx data
+   *
+   * @returns {boolean} true if the coin uses non-packed encoding for tx data
+   */
+  protected decideIfCoinUsesNonPackedEncodingForTxData(): boolean {
```
Comment (mullapudipruthvik) [inline modules/abstract-eth/src/lib/transactionBuilder.ts:639]:
```suggestion
  protected coinUsesNonPackedEncodingForTxData(): boolean {
```

File: modules/sdk-coin-polygon/src/lib/transactionBuilder.ts [hunk 3]
```diff
@@ -26,4 +26,11 @@ export class TransactionBuilder extends EthLikeTransactionBuilder {
     }
     return this._transfer;
   }
+
+  /** @inheritdoc */
+  protected decideIfCoinUsesNonPackedEncodingForTxData(): boolean {
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-polygon/src/lib/transactionBuilder.ts:31]:
```suggestion
  protected coinUsesNonPackedEncodingForTxData(): boolean {
```

File: modules/abstract-eth/src/lib/transactionBuilder.ts (hunk 1 above)
Comment (gianchandania) [reply modules/abstract-eth/src/lib/transactionBuilder.ts:631]: done

File: modules/abstract-eth/src/lib/transactionBuilder.ts (hunk 2 above)
Comment (gianchandania) [reply modules/abstract-eth/src/lib/transactionBuilder.ts:639]: done

**Review by mullapudipruthvik — APPROVED** [review]

**Pull request merged**